### PR TITLE
Soen64 back end update role

### PR DESCRIPTION
**Pull request by jfv-2000**
# Description

-Added a folder for constants such as Roles
-Added protected routes to verify if the user is logged in
-Added protected routes to verify if the user is an admin
-Added the route that allows admins to update a user's role
-Added the Confirmed field in database and back-end to know if a user has been confirmed by the admin, a patient is confirmed by default when signing up.

Fixes [SOEN-64](https://soen390team8.atlassian.net/jira/software/projects/SOEN/boards/1/backlog?selectedIssue=SOEN-64)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] Manually

# Screenshots
Original      | Updated
:--------------:|:--------------:
** original screenshot ** | ** updated screenshot ** |

### CODE REVIEW

**Review by KelvinTrang — APPROVED** [review]
Very nice and clean code. Good job!

**Review by amqkv — APPROVED** [review]
LGTM :D

File: backend/controllers/authController.js [hunk 1]
```diff
@@ -51,9 +56,21 @@ function logIn(req, res) {
...
-    register: register,
-    logIn: logIn
+//Get authenticated user
+function getAuthUser(req, res) {
+    User.findOne({
+        where: {
+          email: req.user.email
+        }
+    }).then(user => {
+          console.log("user: ", user);
```
Comment (amqkv) [inline backend/controllers/authController.js]: [Suggestion] Maybe we could remove unnecessary console.log's for cleaner code

Comment (jfv-2000) [reply backend/controllers/authController.js]: I agree with that ! Did not see this unnecessary console.log, thanks for the suggestion !

File: backend/controllers/adminController.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+function updateRole(req, res) {
+    User.update({
+        Role: req.body.role
+    }, {
+        where: {
+            Email: req.body.email
+        }
+    }).then(result => {
+        console.log("Role successfully updated !");
+        res.json("Role succesfully updated !");
```
Comment (NicholasYip) [inline backend/controllers/adminController.js]: do you think a 200 should be returned also `res.status(200).json("Role succesfully updated !");` or is that not needed?

File: backend/controllers/authController.js [hunk 3]
```diff
@@ -2,19 +2,23 @@ const User = require("../models/user");
...
+        DateOfBirth: req.body.dateOfBirth,
         Address: req.body.address,
         City: req.body.city,
         Email: req.body.email,
         Password: bcrypt.hashSync(req.body.password, 8),
-        PhoneNumber: req.body.phonenumber
+        PhoneNumber: req.body.phoneNumber,
+        PostalCode: req.body.postalCode,
+        Role: req.body.accountRole,
+        Confirmed: req.body.accountRole == constants.ROLE.PATIENT
```
Comment (NicholasYip) [inline backend/controllers/authController.js]: should be `===`

File: backend/utils/constants.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const ROLE = {
+    ADMIN: 'Admin',
+    DOCTOR: 'Doctor',
+    HEALTH_OFFICIAL: 'HealthOfficial',
+    IMMIGRATION_OFFICER: 'ImmigrationOfficer',
+    PATIENT: 'Patient',
+}
```
Comment (NicholasYip) [inline backend/utils/constants.js:7]: love the usage of contants!

**Review by NicholasYip — APPROVED** [review]
Amazing work!

**Review by stevenvuong00 — APPROVED** [review]
looks good to me. good job!

File: backend/controllers/adminController.js (hunk 2 above)
Comment (jfv-2000) [reply backend/controllers/adminController.js]: Excellent idea !

File: backend/controllers/authController.js (hunk 3 above)
Comment (jfv-2000) [reply backend/controllers/authController.js]: Good catch, thanks :)

**Pull request merged**